charms/knative-eventing/tests/unit/test_charm.py

```diff
@@ -1,67 +1,64 @@
 # Copyright 2022 Canonical Ltd.
 # See LICENSE file for licensing details.
-#
-# Learn more about testing at: https://juju.is/docs/sdk/testing
 
-import unittest
-from unittest.mock import Mock
+import pytest
```

Comment:
Thinking about this a bit more... With the resource handler doing a lot of this, maybe there's different ways of testing.  Maybe instead there should be some "change config; execute something that would cause the resource_handler to apply things; catch what it passes to lightkube.Client.apply() and assert whether it is actually correct (eg: it actually has the right config in the right places)`.  This is less an issue for eventing, but more so for serving

Reply:
We can add a more robust suite, there are a couple of test cases that can be added not just to eventing and serving, but also to operator. I opened an [issue](https://github.com/canonical/knative-operators/issues/23) to track that enhancement.